engine/apps/api/views/schedule.py

```diff
@@ -138,13 +139,25 @@ def oncall_users(self):
         since self.get_serializer_context() is called multiple times for every instance in the queryset.
         """
         current_schedules = self.get_queryset(annotate=False).none()
+        events_datetime = datetime.datetime.now(datetime.timezone.utc)
         if self.action == "list":
-            # listing page, only get oncall users for current page schedules
-            current_schedules = self.paginate_queryset(self.filter_queryset(self.get_queryset(annotate=False)))
+            # listing page, only get oncall users for current page schedules, prefetch shift swap requests
+            current_schedules = self.filter_queryset(self.get_queryset(annotate=False)).prefetch_related(
+                Prefetch(
+                    "shift_swap_requests",
+                    queryset=ShiftSwapRequest.objects.filter(
+                        swap_start__lte=events_datetime, swap_end__gte=events_datetime
+                    )
+                    .select_related("benefactor", "beneficiary")
+                    .order_by("created_at"),
+                    to_attr=PREFETCHED_SHIFT_SWAPS,
+                )
+            )
```

Comment:
_nit_ consider extracting out the `Prefetch` construction logic into a staticmethod on the view class to avoid the duplication below